devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -102,6 +102,30 @@ import physicalgraph.zwave.commands.usercodev1.*
 def installed() {
 	// Device-Watch pings if no device events received for 1 hour (checkInterval)
 	sendEvent(name: "checkInterval", value: 1 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	runIn(120, installedCheck)
+}
+
+/**
+ * Verify that we have actually received the lock's initial states.
+ * If not, verify that we have at least requested them or request them,
+ * and check again.
+ */
+def installedCheck() {
+	if (device.currentState("lock") && device.currentState("battery")) {
+		unschedule("installedCheck")
+	} else {
+		// We might have called updated() or configure() at some point but not have received a reply, so don't flood the network
+		if (!state.lastLockDetailsQuery || secondsPast(state.lastLockDetailsQuery, 2 * 60)) {
+			def actions = updated()
+
+			if (actions) {
+				sendHubCommand(actions.toHubAction())
+			}
+		}
+
+		runIn(120, installedCheck)
```

Comment:
Did you test with the DTH running locally? I believe you need to add a third argument `[forceForLocallyExecuting: true]` to get this to run if the DTH is running locally.

Reply:
@tpmanley I will give this a try. Thanks for the heads up! The account I was using with the Connect Home I have doesn't have the account permissions to set the runLocally flag.

Reply:
@tpmanley Update incoming with that change!